internal/compatibility/compatibility.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+/*
+Package compatibility offers helper functions for comparing and filtering versions.
+*/
+package compatibility
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"golang.org/x/mod/semver"
+)
+
+var (
+	// ErrMajorMismatch signals that the major version of two compared versions don't match.
+	ErrMajorMismatch = errors.New("missmatching major version")
+	// ErrMinorDrift signals that the minor version of two compared versions are further apart than one.
+	ErrMinorDrift = errors.New("target version needs to be equal or up to one minor version higher")
+	// ErrSemVer signals that a given version does not adhere to the Semver syntax.
+	ErrSemVer = errors.New("invalid semver")
```

Comment:
```suggestion
	ErrSemVer = errors.New("invalid semantic version")
```
nit: not using abbreviations in error descriptions seems like a good idea